core/src/avm2/activation.rs

```diff
@@ -1247,8 +1278,41 @@ impl<'a, 'gc, 'gc_context> Activation<'a, 'gc, 'gc_context> {
         index: Index<AbcMultiname>,
     ) -> Result<FrameControl<'gc>, Error> {
         let value = self.context.avm2.pop();
-        let multiname = self.pool_multiname(method, index)?;
-        let mut object = self.context.avm2.pop().coerce_to_object(self)?;
+        let txunit = method.translation_unit();
+        let abc = txunit.abc();
+        let abc_multiname = Multiname::resolve_multiname_index(&abc, index.clone())?;
+        let (multiname, mut object) = if matches!(
+            abc_multiname,
+            AbcMultiname::MultinameL { .. } | AbcMultiname::MultinameLA { .. }
```

Comment:
(I feel like some of this code could be deduplicated between these three opcodes, but this can wait till we extend it more, like with the int-path.)